modeler.js

```diff
@@ -0,0 +1,144 @@
+import axios from "axios";
+import { getAccessToken } from "./auth.js";
+
+const authorizationConfiguration = {
+  clientId: process.env.MODELER_CLIENT_ID,
+  clientSecret: process.env.MODELER_CLIENT_SECRET,
+  audience: process.env.MODELER_AUDIENCE
+};
+
+// An action that creates a new project.
+async function createProject([projectName]) {
+  // Every request needs an access token.
+  const accessToken = await getAccessToken(authorizationConfiguration);
+
+  // These settings come from your .env file.
+  const modelerApiUrl = process.env.MODELER_BASE_URL;
+
+  // This is the API endpoint to create a new project.
+  const url = `${modelerApiUrl}/projects`;
+
+  // Configure the API call.
+  const options = {
+    method: "POST",
+    url,
+    headers: {
+      Accept: "application/json",
+      Authorization: `Bearer ${accessToken}`
+    },
+    data: {
+      // The body contains information about the new project.
+      name: projectName
+    }
+  };
+
+  try {
+    // Call the add endpoint.
+    const response = await axios(options);
+
+    // Process the results from the API call.
+    const newProject = response.data;
+
+    console.log(
+      `Project added! Name: ${newProject.name}. ID: ${newProject.id}.`
+    );
+  } catch (error) {
+    // Emit an error from the server.
+    console.error(error.message);
+  }
+}
+
+// A function that adds a new collaborator to a project
+async function addCollaborator([collaboratorEmail, projectId, role]) {
+  // Every request needs an access token.
+  const accessToken = await getAccessToken(authorizationConfiguration);
+
+  // These settings come from your .env file.
+  const modelerApiUrl = process.env.MODELER_BASE_URL;
+
+  // This is the API endpoint to add a new collaborator.
+  const url = `${modelerApiUrl}/collaborators`;
+
+  // Configure the API call.
+  const options = {
+    method: "PUT",
+    url,
+    headers: {
+      Accept: "application/json",
+      Authorization: `Bearer ${accessToken}`
+    },
+    data: {
```

Comment:
I had a tough time ensuring I got this data and the response data correct, but I think I'm close.

Reply:
Did you have problems because you can't test/run this?

> Adds a new collaborator to a project or modifies the permission level of an existing collaborator.

> Note: Only users that are part of the authorized organization (see GET /api/v1/info) and logged in to Web Modeler at least once can be added to a project.

After reading this in the docs, does it make sense to include it with the tutorial? While we could use this as a teachable moment and let the call fail, I'm concerned about the experience of an individual doing an evaluation. Could they easily create a test user to see this collaboration work in action?

Reply:
Ah, I see.

Hmm...I'm not sure what API calls we could make for this tutorial for a beginner without these little road blocks.

The alternative is to add disclaimers/admonitions throughout the tutorial to be clear on this, or require a larger number of prerequisites.

I'll do some more digging, and sync with @pepopowitz at the retreat on options.

Reply:
After taking another pass through the Modeler API docs, I feel good about the requests you've put together here. 

The only thing I might add is a "deleteProject" at the end, so that we don't pollute their environment with test data. I don't feel strongly enough to push back on that in a review, but if it resonates and you think it's worthwhile, cool.

I'm now going to take a walk through the tutorial you've written!

Reply:
hmm, I guess you haven't updated the tutorial docs based on these changes, so I'll just see if I can make this code all run.

Reply:
@pepopowitz Since we were a bit back and forth on the changes, I wanted to hold off on the docs changes until we were confident. Will update once this PR is finalized 👍